### DISCUSSION TITLE
Disable introspection query with spring boot using introspection-enabled property

### DISCUSSION THREAD

**Discussion by JalajChawla** [graphql-java-kickstart/graphql-spring-boot#800] [Q&A]
I am facing issues while disabling the introspection query in a spring boot graphql project on a get endpoint query parameter.

I was replicating this by using one of the GET endpoint and using the below parameter 

`baseurl/servicename/insight_graph?query=fragment+FullType+on+__Type+{++kind++name++description++fields(includeDeprecated%3a+true)+{++++name++++description++++args+{++++++...InputValue++++}++++type+{++++++...TypeRef++++}++++isDeprecated++++deprecationReason++}++inputFields+{++++...InputValue++}++interfaces+{++++...TypeRef++}++enumValues(includeDeprecated%3a+true)+{++++name++++description++++isDeprecated++++deprecationReason++}++possibleTypes+{++++...TypeRef++}}fragment+InputValue+on+__InputValue+{++name++description++type+{++++...TypeRef++}++defaultValue}fragment+TypeRef+on+__Type+{++kind++name++ofType+{++++kind++++name++++ofType+{++++++kind++++++name++++++ofType+{++++++++kind++++++++name++++++++ofType+{++++++++++kind++++++++++name++++++++++ofType+{++++++++++++kind++++++++++++name++++++++++++ofType+{++++++++++++++kind++++++++++++++name++++++++++++++ofType+{++++++++++++++++kind++++++++++++++++name++++++++++++++}++++++++++++}++++++++++}++++++++}++++++}++++}++}}query+IntrospectionQuery+{++__schema+{++++queryType+{++++++name++++}++++mutationType+{++++++name++++}++++types+{++++++...FullType++++}++++directives+{++++++name++++++description++++++locations++++++args+{++++++++...InputValue++++++}++++}++}}`

and with the below dependencies

```
        <dependency>
            <groupId>com.graphql-java-kickstart</groupId>
            <artifactId>graphql-spring-boot-starter</artifactId>
            <version>5.4.1</version>
        </dependency>

        <dependency>
            <groupId>com.graphql-java-kickstart</groupId>
            <artifactId>graphql-java-tools</artifactId>
            <version>5.4.1</version>
            <exclusions>
                <exclusion>
                    <artifactId>jackson-module-kotlin</artifactId>
                    <groupId>com.fasterxml.jackson.module</groupId>
                </exclusion>
            </exclusions>
        </dependency>

        <dependency>
            <groupId>com.graphql-java-kickstart</groupId>
            <artifactId>graphiql-spring-boot-starter</artifactId>
            <version>5.4.1</version>
        </dependency>
```
        
This works pretty well when we request the same using post and with the below request body 
        
      
 POST baseurl/servicename/insight_graph
        
        [
    {
        "operationName": "IntrospectionQuery",
        "variables": {},
        "query": "query IntrospectionQuery {__schema {queryType { name },mutationType { name },subscriptionType { name },types {...FullType},directives {name,description,args {...InputValue},onOperation,onFragment,onField}}}\nfragment FullType on __Type {kind,name,description,fields(includeDeprecated: true) {name,description,args {...InputValue},type {...TypeRef},isDeprecated,deprecationReason},inputFields {...InputValue},interfaces {...TypeRef},enumValues(includeDeprecated: true) {name,description,isDeprecated,deprecationReason},possibleTypes {...TypeRef}}\nfragment InputValue on __InputValue {name,description,type { ...TypeRef },defaultValue}\nfragment TypeRef on __Type {kind,name,ofType {kind,name,ofType {kind,name,ofType {kind,name}}}}"
    }
]

I also tried this via spring filters and it works fine.
Is there any way to disable introspection via spring boot property for this?

**Comment by oliemansm** [comment]
Converted to Discussion, since this is a general question, not a bug. You should upgrade to latest versions of the libraries and read the README. There's a property described to enable or disable the introspection query there: https://github.com/graphql-java-kickstart/graphql-spring-boot#graphql-java-tools.

```yaml
graphql:
  tools:
    # Enable or disable the introspection query. Disabling it puts your server in contravention of the GraphQL
    # specification and expectations of most clients, so use this option with caution
    introspection-enabled: false
```
